### PR TITLE
feat: add fused_remap_deepep Triton kernel for DeepEP shared expert fusion

### PR DESCRIPTION
**Pull request by syy-hw**
## Summary
- Add `fused_remap_deepep` Triton kernel under `moe/` — fuses routed topk expansion with shared experts and remaps IDs to DeepEP interleaved layout
- Decoupled from sglang internals: `ep_rank`, `ep_size`, `routed_scaling_factor` are passed as explicit parameters instead of importing from sglang's `parallel_state`
- Include unit tests covering: basic case (1 shared expert), multiple shared experts, and empty batch

## Context
Extracted from [sglang PR #26155](https://github.com/sgl-project/sglang/pull/26155) (NPU DeepEP fused shared experts). The kernel was originally inline in sglang's `topk.py` and is being moved to sgl-kernel-npu following the project convention of keeping kernel code in the independent repo.

### CODE REVIEW

File: python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        topk_ids: [N, K] routed expert IDs (int).
+        topk_weights: [N, K] routed expert weights (float).
+        num_fused_shared_experts: number of shared experts to fuse.
+        num_physical_routed_experts: total physical routed experts across EP.
+        ep_rank: expert parallel rank of this worker.
+        ep_size: expert parallel world size.
+        routed_scaling_factor: scaling factor for routed experts (0 or 1 means no scaling).
+    """
+    if topk_ids.shape[0] == 0:
+        return topk_ids, topk_weights
```
Comment (AndyLi429) [inline python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py]: Empty-batch early return has wrong output shape?returns tensors with shape [0, K], but callers expect shape [0, K + num_fused_shared_experts]

File: python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    num_local_routed = num_physical_routed_experts // ep_size
+    num_local_experts = num_local_routed + num_fused_shared_experts
+
+    N = topk_ids.shape[0]
+    K = topk_ids.shape[1]
+    K_PLUS_N = K + num_fused_shared_experts
```
Comment (AndyLi429) [inline python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py]: in the same kernel N already means "number of rows/tokens."
K_PLUS_N has confused meaning

File: python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py (hunk 1 above)
Comment (syy-hw) [reply python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py]: Good catch! Fixed in fcc82df — the early return now produces tensors with the correct output shape (0, K + num_fused_shared_experts). Also updated the empty-batch test to assert on the full shape (0, 7) instead of only checking shape[0].

File: python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py (hunk 2 above)
Comment (syy-hw) [reply python/sgl_kernel_npu/sgl_kernel_npu/moe/fused_remap_deepep.py]: Agreed — K_PLUS_N was confusing since N already means "number of rows" in the kernel. Renamed to TOTAL_COLS for clarity. See fcc82df.
